sdk/core/azure-core/src/main/java/com/azure/core/implementation/util/FileContent.java

```diff
@@ -146,11 +188,14 @@ public int getChunkSize() {
     }
 
     private byte[] getBytes() {
-        try {
-            return Files.readAllBytes(file);
-        } catch (IOException exception) {
-            throw LOGGER.logExceptionAsError(new UncheckedIOException(exception));
+        if (length > MAX_ARRAY_SIZE) {
```

Comment:
Love it! Rather throw early here when we know we'll fail. But I'd have this throw an OutOfMemoryError which is consistent with how the JDK handles scenarios like these.

Reply:
Good point, and maybe in this case `IllegalStateException` instead of `IllegalArgumentException`?